src/components/layout/header/index.tsx

```diff
@@ -34,7 +34,11 @@ export const Header = () => {
   const { navigationSitemapShowing } = useAppStore()
   const toggle = useToggleState()
   const router = useRouter()
+  const [platform, setPlatform] = useState<string | null>(null)
 
+  useEffect(() => {
+    setPlatform(getPlatform())
```

Comment:
:hand: Ideally we wouldn't even need the state here, because we _ought_ to be able to just derive this value while rendering.

However, Next is also going to render the _initial_ HTML on the server, and the server has no idea what the user's browser is.  (Mostly. It could figure it out by looking at the HTTP request, but we don't have access to that in this particular file.)

So yes, it's reasonable to have state here, and immediately set that state as soon as this component renders on the client.

To avoid a small flash, we should call `useLayoutEffect()` rather than `useEffect()`, since `useLayoutEffect` runs in the commit phase before the browser has a chance to paint.

Change that and this should be okay, I think.